File: src/main/java/com/refinedmods/refinedstorage/api/storage/IStorageProvider.java
package com.refinedmods.refinedstorage.api.storage;

import net.minecraft.world.item.ItemStack;
import net.neoforged.neoforge.fluids.FluidStack;
import java.util.List;

/**
 * Represents a node that provides the network with storage.
 */
public interface IStorageProvider {
    /**
     * @param storages the item storages
     */
    void addItemStorages(List<IStorage<ItemStack>> storages);

    /**
     * @param storages the fluid storages
     */
    void addFluidStorages(List<IStorage<FluidStack>> storages);
}

File: src/main/kotlin/com/refinedmods/refinedstorage/api/storage/IStorageProvider.kt
package com.refinedmods.refinedstorage.api.storage

import net.minecraft.item.ItemStack
import reborncore.common.fluid.container.FluidInstance


/**
 * Represents a node that provides the network with storage.
 */
interface IStorageProvider {
    /**
     * @param storages the item storages
     */
    fun addItemStorages(storages: List<IStorage<ItemStack>>)

    /**
     * @param storages the fluid storages
     */
    fun addFluidStorages(storages: List<IStorage<FluidInstance>>)
}